src/types.ts

```diff
@@ -0,0 +1,11 @@
+export interface AuthBody {
+  email: string;
+  password: string;
+  name?: string;
+  lastName?: string;
+}
+export interface CategoryType {
+  _id?: string;
```

Comment:
If _id is required then don't add optional property in the category type. Apart from this all looks good. Good folder structure.

Reply:
i am using categoryType in other objs as well where id is not require should i make separate type for that or i can use like this